cluster-topology.md

```diff
@@ -0,0 +1,273 @@
+
+## Scale out control plane nodes
+
+A common Kubernetes cluster maintenance activity is scaling out (or in) the number of control plane nodes in response to cluster activity. Because Cluster API configuration interfaces are in fact Kubernetes resources, there are a lot of ways to do this. We'll demonstrate using a variety of methods.
+
+### Scale from 1 to 3 control plane nodes
+
+Because we are leveraging the Kubernetes declarative model, we can simply refer to a desired configuration specification and rely upon Cluster API to evaluate what's different between the two.
+
+Assuming that your "`docker-cluster-one`" Cluster still has its original configuration, you should have one control plane node:
+
+```sh
+kubectl --kubeconfig cluster-one.kubeconfig get nodes
+```
+
+Output:
+
+```
+NAME                                             STATUS   ROLES           AGE   VERSION
+docker-cluster-one-md-0-nrh7k-75ddc4778f-vlph9   Ready    <none>          24m   v1.24.6
+docker-cluster-one-xbqb2-tlmpb                   Ready    control-plane   24m   v1.24.6
+```
+
+Let's use the idempotent model to submit a modified configuration of our "`docker-cluster-one`" Cluster with 3 control plane replicas instead of 1. We've provided [a reference yaml of this updated configuration in this repo](yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml).
+
+If you're on masOS or Linux you can diff this modified spec from the original spec used to create the "`docker-cluster-one`" Cluster:
+
+```sh
+diff yamls/clusters/docker-cluster-one.yaml yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml
+```
+
+Output:
+
+```
+17c17
+<       replicas: 1
+---
+>       replicas: 3
+```
+
+The above shows that the `yaml` specs are almost identical, with the only change being the `replicas` value on L17. By applying that modified spec to our kind management cluster we can achieve a control plane node scale out operation:
+
+```sh
+kind get kubeconfig --name kubecon-na-22-capi-lab > kubecon-na-22-capi-lab.kubeconfig
+kubectl --kubeconfig kubecon-na-22-capi-lab.kubeconfig apply -f yamls/clusters/docker-cluster-one-3-control-plane-replicas.yaml
+```
+
+Output:
+
+```
+cluster.cluster.x-k8s.io/docker-cluster-one configured
+```
+
+Now we can watch the new control plane nodes come online:
+
+```sh
+kubectl --kubeconfig cluster-one.kubeconfig get nodes -w
+```
+
+An interesting note! As your cluster transitions from 1 to 2 control plane nodes, you will temporarily lost etcd quorum and your the apiserver running on your cluster will briefly go offline. So your `kubectl -w` command will be interrupted. Re-run the same command again to resume watching your cluster nodes:
```

Comment:
```suggestion
An interesting note! As your cluster transitions from 1 to 2 control plane nodes, it will temporarily lose etcd quorum and the apiserver running on your cluster will briefly go offline. So your `kubectl -w` command will be interrupted. Re-run the same command again to resume watching your cluster nodes:
```
I think

Reply:
thx for the English lesson :)